SVCPLAN-7531: Add an IB performance script to admin tools

inwho opened the pull request:
Currently used on Magnus, modified for hopefully better generalized use.

Tested the updated script on mg-adm01 with the same test suite of configuration files we use during PMs.

inwho committed:
SVCPLAN-7531: Add an IB performance script to admin tools

jakerundall commented on admin_scripts/ib_performance.sh:
```diff
@@ -0,0 +1,98 @@
+    d) DEV=${OPTARG};;
+    p) port=${OPTARG};;
+    s) size=${OPTARG};;
+    c) source ${OPTARG};;
```
That's a reasonable suggestion (even though the file is unlikely to have a space in it or anything like that).

jakerundall commented on conf/ib_performance.conf.example:
```diff
@@ -0,0 +1,30 @@
+HOST=test001
+RANGE=test[201-210]
+VALUE=5000
+port=55555
```
I think Copilot also had the suggestion to change port to PORT here in the example config file. That also seems reasonable to me, but if you do I think you'd need to do so in the actual script as well.

And if you were going to do that, I guess I might consider the same for size?

jakerundall commented on conf/ib_performance.conf.example:
```diff
@@ -0,0 +1,30 @@
+RANGE=lnet
+VALUE=10000
+TAIGA=true
+port=988
```
(So port -> PORT here too, if you choose to do so.)

jakerundall commented on admin_scripts/ib_performance.sh:
```diff
@@ -0,0 +1,98 @@
+#
+# -V, VALUE	A threshold value (MB/sec) of expected performance. Can be used to filter results to show only underperforming nodes
+#
+# -T, TAIGA	Flag for taiga side testing. Prepends "tg-" to all nodes in RANGE e.g. test001 -> tg-test001
```
I think we'd discussed this in-person, but one thought was that making a Taiga-specific flag might be a little more specific than we want, because we could simply have a separate config file for the Taiga-side testing.

inwho replied:
Thanks for the reminder, I had forgotten through the multiple iterations of the script.
I've abstracted the functionality to allow targeting of a specific receiver on HOST through use of TARGET, which we can store in the config file. Unfortunately I couldn't quite get the IB connections to work without the additional parameter.

jakerundall commented on admin_scripts/ib_performance.sh:
```diff
@@ -0,0 +1,98 @@
+# -p, <port>	Listen on/connect to port <port> (default is 18515)
+#
+# -s, <size>	Size of message to exchange (default 65536)
+#
```
For consistency, consider formatting <dev>, <port>, and <size> as DEV, PORT, and SIZE.

jakerundall commented on admin_scripts/ib_performance.sh:
```diff
@@ -0,0 +1,98 @@
+    T) TAIGA=true;;
+    d) DEV=${OPTARG};;
+    p) port=${OPTARG};;
+    s) size=${OPTARG};;
```
Consider port->PORT and size-> SIZE.

jakerundall commented on admin_scripts/ib_performance.sh:
```diff
@@ -0,0 +1,98 @@
+  flag_params+=" -p $port"
+fi
+if [[ -n "${size}" ]]; then
+  flag_params+=" -s $size"
```
Consider port->PORT and size->SIZE.

jakerundall reviewed (commented):
See in-inline comments. Thanks!

inwho commented:
I've updated the script with the recommendations to standardize parameters by making them all UPPER CASE, and also took the chance to standardize the flags and make them all lower case. 

Reworked the TAIGA boolean to the more generalized TARGET string, which allows specification of a receiver on HOST. This allows for a more flexible use of the parameter which also includes the original use case.

Updated the in progress message to specify what two endpoints an IB connection is being made between. This should help with troubleshooting, and clarifying what is happening within the script.

jakerundall reviewed (approved):
Looks good!
